### PR TITLE
Restrict numeric workarounds to ROCm GPUs

### PR DESCRIPTION
**Pull request by amorehead**
* Updates and restricts numeric workarounds to ROCm GPUs.
* As an aside, how important is it that the [`softmax` operation](https://github.com/lucidrains/alphafold3-pytorch/blob/097cbff6beb1df51a9c3bbc8cf0a2e2043c9cb9d/alphafold3_pytorch/attention.py#L510) be in `torch.float32` precision (e.g., compared to being autocast to `bfloat16`)? I ask because for large (5000+ atom) inputs, float32 precision requires 30GB or more of GPU memory to compute this softmax when using several `diffusion_num_augmentations`.

### CODE REVIEW

**Comment by lucidrains** [comment]
@amorehead hey, i think bfloat16 is acceptable, let me make that casting customizable.

this precaution was done for float16

**Comment by lucidrains** [comment]
@amorehead man, you are truly in the wild west

**Comment by amorehead** [comment]
Tell me about it

**Comment by amorehead** [comment]
> @amorehead hey, i think bfloat16 is acceptable, let me make that casting customizable.
> 
> this precaution was done for float16

On my side, I just removed the `dtype=torch.float32` bit in `attention.py` so that torch's AMP package can decide which precision to use there.

**Comment by lucidrains** [comment]
@amorehead yup that works, i just made it customizable in the latest version, and defaulted the casting to off
